packages/sdk/src/core/teamsfx.ts

```diff
@@ -28,6 +29,40 @@ const ReservedKey: Set<string> = new Set<string>([
   "sqlIdentityId",
 ]);
 
+/**
+ * Frank edit
+ * OBOUserCredentialClientSecretConfig, OBOUserCredentialCertificateContentConfig, AppCredentialConfig classes
+ */
+export type OBOUserCredentialClientSecretConfig = {
+  [key: string]: string,
+  "clientId": string,
+  "authorityHost": string,
+  "clientSecret": string,
+  "tenantId": string
+}
+
+export type OBOUserCredentialCertificateContentConfig = {
+  [key: string]: string,
+  "clientId": string,
+  "authorityHost": string,
+  "certificateContent": string,
+  "tenantId": string
+}
+
+export type AppCredentialClientSecretConfig = {
+  [key: string]: string,
+  "clientId": string,
+  "clientSecret": string,
+  "tenantId": string
+}
+
+export type AppCredentialCertificateContentConfig = {
+  [key: string]: string,
+  "clientId": string,
+  "certificateContent": string,
+  "tenantId": string
```

Comment:
same here for all the property should be optional, since it can use environment variable by default